docs/proposals/TLSPolicy.md

```diff
@@ -0,0 +1,84 @@
+# WIP
+# TLSPolicy API
+Currently, TLS certificates issued by the MGC are issued and managed using `cert-manager`. Configuration for these certificates are opaque to the end-user - we'd like to gradually expose a TLSPolicy to cover a range of use cases where administrators require more flexible configuration.
+
+## Problem
+
+Gateway Administrators require a comprehensive way to manage Automated, ACME-based TLS for Gateways. As gateways multiply and scale, there's a need for a flexible, policy-based API for handling Automatic TLS. Currently (July 2023), the MGC sets up automated TLS whenever a Gateway listener is created. There are currently no hooks into the creation of automated TLS certificates, nor user-exposed configuration options in terms ACME providers, custom renewal windows/schedules, key size, algorithm and other configuration options available via `cert-manager`.
+
+## Use Cases
+
+**Choosing a TLS Provider**: As an admin, I want to select a specific ACME provider for the TLS certificates in a particular gateway.
```

Comment:
Should this be more focused on Cert Manager Issuers that ACME providers? The majority of issuers are more than likely ACME, but not all of them. Self Signed cert issuer for instance aren't ACME.

Reply:
Good point. I think it should be more focused on cert-manager issuers, rather than plain ACME ones as you say.